### PR TITLE
kernel: update to 5.10.88.1

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Update kernel to 5.10.88.1 to fix CVE 2021-43975, CVE-2021-45480, CVE-2021-45486
Add patch for kernel CVE 2021-43976
###### Change Log  

- Update kernel source to 5.10.88.1
- Addresses kernel CVEs: CVE-2021-43975, CVE-2021-43976, CVE-2021-45480, CVE-2021-45486

###### Does this affect the toolchain?  

YES, kernel-headers
###### Associated issues  

none

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-43975
- https://nvd.nist.gov/vuln/detail/CVE-2021-43976
- https://nvd.nist.gov/vuln/detail/CVE-2021-45480
- https://nvd.nist.gov/vuln/detail/CVE-2021-45486

###### Test Methodology

- Pipeline build id AMD: 156466
- Pipeline build id ARM: 156882
- local build

### CODE REVIEW

File: SPECS/kernel/kernel.spec [hunk 1]
```diff
@@ -572,6 +575,10 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 
 
 %changelog
+* Mon Jan 3 2022 Cameron Baird <cameronbaird@microsoft.com> - 5.10.88.1-1
+- Update Kernel source to 5.10.88.1
+- Fixes CVE-43975
```
Comment (rlmenge) [inline SPECS/kernel/kernel.spec:580]: Please mention both CVEs and create a new line for the patch
- Addressed CVE-2021-43975
- Patch for CVE-2021-43976

File: SPECS/kernel-headers/kernel-headers.spec [hunk 2]
```diff
@@ -39,6 +41,10 @@ cp -rv usr/include/* /%{buildroot}%{_includedir}
 %{_includedir}/*
 
 %changelog
+* Mon Jan 3 2022 Cameron Baird <cameronbaird@microsoft.com> - 5.10.88.1-1
+- Update Kernel source to 5.10.88.1
+- Fixes CVE-43975
```
Comment (rlmenge) [inline SPECS/kernel-headers/kernel-headers.spec:46]: The Fixes line isn't needed here, just mention why a patch was applied

File: SPECS-SIGNED/kernel-signed/kernel-signed.spec [hunk 3]
```diff
@@ -147,6 +147,10 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %endif
 
 %changelog
+* Mon Jan 3 2022 Cameron Baird <cameronbaird@microsoft.com> - 5.10.88.1-1
+- Update Kernel source to 5.10.88.1
+- Fixes CVE-43975
```
Comment (rlmenge) [inline SPECS-SIGNED/kernel-signed/kernel-signed.spec:152]: Fixes line not needed

File: SPECS/hyperv-daemons/hyperv-daemons.spec [hunk 4]
```diff
@@ -223,6 +224,10 @@ fi
 %{_sbindir}/lsvmbus
 
 %changelog
+* Mon Jan 3 2022 Cameron Baird <cameronbaird@microsoft.com> - 5.10.88.1-1
+- Update Kernel source to 5.10.88.1
+- Fixes CVE-43975
```
Comment (rlmenge) [inline SPECS/hyperv-daemons/hyperv-daemons.spec:229]: Fixes line not needed , please specify why patch was needed

File: SPECS/hyperv-daemons/hyperv-daemons.spec [hunk 5]
```diff
@@ -108,7 +109,7 @@ Contains tools and scripts useful for Hyper-V guests.
 %setup -q -n CBL-Mariner-Linux-Kernel-rolling-lts-mariner-%{version}
 %patch0 -p1
 %patch1 -p1
-
```
Comment (rlmenge) [inline SPECS/hyperv-daemons/hyperv-daemons.spec:111]: Please add newline back just for consistency

File: SPECS/kernel-hyperv/kernel-hyperv.spec [hunk 6]
```diff
@@ -271,6 +273,10 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %{_libdir}/perf/include/bpf/*
 
 %changelog
+* Mon Jan 3 2022 Cameron Baird <cameronbaird@microsoft.com> - 5.10.88.1-1
+- Update Kernel source to 5.10.88.1
+- Fixes CVE-43975
```
Comment (rlmenge) [inline SPECS/kernel-hyperv/kernel-hyperv.spec:278]: same as other comments

**Review by rlmenge — CHANGES_REQUESTED** [review]

File: SPECS/kernel/kernel.spec [hunk 7]
```diff
@@ -18,6 +18,7 @@ Source3:        sha512hmac-openssl.sh
 Source4:        cbl-mariner-ca-20210127.pem
 Patch0:         0001-clocksource-drivers-hyper-v-Re-enable-VDSO_CLOCKMODE.patch
 Patch1:         0002-add-linux-syscall-license-info.patch
+Patch2:         0003-CVE-2021-43976.patch
```
Comment (christopherco) [inline SPECS/kernel/kernel.spec:21]: @nisamson will our patch tooling pick up that CVE-2021-43976 has been patched if we have the `0003-` prefix?

Comment (Camelron) [reply SPECS/kernel/kernel.spec:21]: Is there a preferred naming convention for CVE patches that actually do something? I.e. not a CVE-2012-12345.nopatch?

**Comment by Camelron** [comment]
@rlmenge 
The reason CONFIG_I8K is removed in kernel-hyperv/config and not kernel/config is that the update from 78-->88 made I8K depends on CONFIG_HWMON, which is set as a module in kernel but is explicitly unset in kernel-hyperv. 

Similarly, in going from 78-->88 CONFIG_HID_CHICONY and CONFIG_HID_SAMSUNG starts depending on USB_HID. kernel/config has USB_HID=m, kernel-hyperv/config omits USB_HID.

Did some more digging to figure out what config USB_HID is: "Say Y here if you want to connect USB keyboards,
	  mice, joysticks, graphic tablets, or any other HID based devices
	  to your computer via USB, as well as Uninterruptible Power Supply
	  (UPS) and monitor control devices."

File: SPECS/kernel/CVE-2021-43975.nopatch [hunk 8]
```diff
@@ -0,0 +1,3 @@
+CVE-2021-43975 - already patched in 5.10.88.1 stable kernel
+Upstream: 2c514d25003ac89bb7716bb4402918ccb141f8f5
```
Comment (rlmenge) [inline SPECS/kernel/CVE-2021-43975.nopatch:2]: These are switched
Upstream: b922f622592af76b57cbc566eaeccda0b31a3496
Stable: 2c514d25003ac89bb7716bb4402918ccb141f8f5

**Review by rlmenge — CHANGES_REQUESTED** [review]

File: SPECS/kernel/kernel.spec (hunk 7 above)
Comment (Camelron) [reply SPECS/kernel/kernel.spec]: For now, I'll just drop the 000X- naming convention.

**Review by rlmenge — APPROVED** [review]
:shipit:

File: SPECS/kernel/kernel.spec [hunk 9]
```diff
@@ -570,8 +576,12 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %{_sbindir}/bpftool
 %{_sysconfdir}/bash_completion.d/bpftool
 
-
 %changelog
+* Mon Jan 03 2022 Cameron Baird <cameronbaird@microsoft.com> - 5.10.88.1-1
+- Update Kernel source to 5.10.88.1
+- Addressed CVE-2021-43975
```
Comment (rlmenge) [inline SPECS/kernel/kernel.spec:582]: Please add new nopatches

**Review by rlmenge — CHANGES_REQUESTED** [review]

**Review by rlmenge — APPROVED** [review]

**Pull request merged**